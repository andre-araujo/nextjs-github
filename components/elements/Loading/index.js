import { Component } from 'react';

import Router from 'next/router';
import Wrapper from './styles';

class Loading extends Component {
    state = {
        isActive: false,
    }

    componentDidMount() {
        Router.onRouteChangeStart = () => this.setState({ isActive: true });
        Router.onRouteChangeComplete = () => this.setState({ isActive: false });
        Router.onRouteChangeError = () => this.setState({ isActive: false });
    }

    render() {
        return (
            <Wrapper active={this.state.isActive}>
                <span />
                <span />
                <span />
            </Wrapper>
        );
    }
}

export default Loading;
