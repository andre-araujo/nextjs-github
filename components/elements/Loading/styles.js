import styled, { css, keyframes } from 'styled-components';

const animateDot = keyframes`
    0% {
        opacity: 0;
        transform: rotate(-90deg);
    }

    50% {
        opacity: 1;
        transform: rotate(0);
    }
`;

export default styled.div`
    align-items: center;
    background: rgba(255, 255, 255, 0.85);
    bottom: 0;
    display: flex;
    left: 0;
    opacity: 0;
    pointer-events: none;
    position: fixed;
    justify-content: center;
    right: 0;
    top: 0;
    z-index: 10;

    ${({ active }) => active && css`
        opacity: 1;
        pointer-events: all;
    `}

    span {
        animation: ${animateDot} 1s infinite;
        background: #666;
        display: inline-flex;
        height: 1rem;
        margin: 0.5rem;
        opacity: 0;
        transform: rotate(90deg);
        width: 1rem;
    }

    span:nth-child(2) {
        animation-delay: 0.1s;
    }

    span:nth-child(3) {
        animation-delay: 0.2s;
    }
`;
