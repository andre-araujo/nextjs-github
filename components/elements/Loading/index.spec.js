import React from 'react';
import { shallow } from 'enzyme';

import Loading from './index';

describe('Loading component', () => {
    it('should match snapshot', () => {
        expect(shallow(
            <Loading />,
        )).toMatchSnapshot();
    });
});
