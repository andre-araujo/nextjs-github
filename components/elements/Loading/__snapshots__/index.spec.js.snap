// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component should match snapshot 1`] = `
<styled.div
  active={false}
>
  <span />
  <span />
  <span />
</styled.div>
`;
